ggml-sycl.cpp

```diff
@@ -15276,24 +15279,38 @@ static void ggml_sycl_mul_mat_batched_sycl(const ggml_tensor *src0,
 
     dpct::library_data_t cu_compute_type = dpct::library_data_t::real_float;
     dpct::library_data_t cu_data_type = dpct::library_data_t::real_float;
+    if (no_mixed_dtypes) {
```

Comment:
@abhilash1910  I think in that case the `no_mixed_dtypes` should be a compile time value, a `constexpr` as well or a template parameter maybe, otherwise it wouldn't compile.

Reply:
The backend is only known at runtime, this would not be valid code

Reply:
In that case, it is not needed.